transports/ftp/src/test/java/org/mule/transport/ftp/FtpClientDestructionOnSocketTimeoutTestCase.java

```diff
@@ -0,0 +1,229 @@
+
+    private static long SERVER_SLEEP = parseLong(CONNECTION_TIMEOUT) + SERVER_DELTA;
+
+    private static Latch serverLatch;
+
+    private final String nameScenario;
+
+    private static FtpConnector connectorSpy;
+
+    private boolean ftpClientWasReleased;
+
+    private static AtomicBoolean alreadySentAListCommand = new AtomicBoolean(false);
+
+    public FtpClientDestructionOnSocketTimeoutTestCase(Ftplet ftpLet, String nameScenario)
+    {
+        super(FLOW, "ftp-connection-timeout-config-flow.xml");
+        setStartContext(false);
+        this.ftplet = ftpLet;
+        this.nameScenario = nameScenario;
+    }
+
+    @Override
+    protected void doSetUp() throws Exception
+    {
+        serverLatch = new Latch();
+        ftpClientWasReleased = false;
+        alreadySentAListCommand.set(false);
+        super.doSetUp();
+    }
+
+    @After
+    public void tearDown() throws Exception
+    {
+        serverLatch.countDown();
+    }
+
+    // Test construction parameters
+    @Parameters
+    public static Collection<Object[]> data()
+    {
+        List<Object[]> parameters = new ArrayList<>();
+        parameters.add(new Object[] { ftpLetOnListCommandSleep, "LIST Command executed Scenario"});
+        return parameters;
+    }
+
+    @Test
+    public void testDestroyFtpClientOnTimeoutConnection() throws Exception
+    {
+        muleContext.start();
+
+        // Wait for FtpReceiver to start polling
+        assertListCommandIsIssued();
+
+        // Intercept releaseFtp method invocation on FtpConnector object, in order to verify that
+        // the FtpClient is destroyed
+        Mockito.doAnswer(new Answer()
+        {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable
+            {
+                // 'releaseFtp' methods second parameter decides whether the FtpClient should be destroyed
+                ftpClientWasReleased = invocation.getArgumentAt(2, boolean.class);
+                return null;
+            }
+        }).when(connectorSpy).releaseFtp(Mockito.any(DefaultInboundEndpoint.class), Mockito.any(FTPClient.class), Mockito.anyBoolean());
+
+        // Since already a LIST command was seen, in the second one, the socket timeout should occur
+        new ConnectionListener(muleContext)
+                .setNumberOfExecutionsRequired(1)
+                .setExpectedAction(CONNECTION_FAILED)
+                .waitUntilNotificationsAreReceived();
+
+        assertThat(alreadySentAListCommand(), is(true));
+        assertThat(ftpClientWasReleased, is(true));
+    }
+
+    private void assertListCommandIsIssued()
+    {
+        PollingProber prober = new PollingProber(WAIT_TIMEOUT, TEST_DELTA);
+        prober.check(new Probe()
+        {
+            @Override
+            public boolean isSatisfied()
+            {
+                return FtpClientDestructionOnSocketTimeoutTestCase.alreadySentAListCommand();
```

Comment:
static import

Reply:
This method should be invoked like this, since its being called from an inner class declaration.

Reply:
Ok.